### PR TITLE
1048 action create/edit gitlab epics

### PR DESCRIPTION
**Pull request by luancazarine**
Actions
  - Create Epic
  - Update Epic

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/gitlab/gitlab.app.mjs [hunk 1]
```diff
@@ -168,6 +217,45 @@ export default {
...
+      label: "Color",
+      description: "The color of the epic. Introduced in GitLab 14.8, behind a feature flag named epic_highlight_color (disabled by default)",
+      optional: true,
+    },
+    confidential: {
+      type: "boolean",
+      label: "Confidential",
+      description: "Whether the epic should be confidential",
+      default: true,
+      optional: true,
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:231]: I think that if it is `optional` the `default` shouldn't be set

File: components/gitlab/gitlab.app.mjs [hunk 2]
```diff
@@ -168,6 +217,45 @@ export default {
...
+      default: false,
+      description: "Whether start date should be sourced from start_date_fixed or from milestones (in GitLab 11.3 and later)",
+      optional: true,
+    },
+    due_date_is_fixed: {
+      type: "boolean",
+      label: "Due date is fixed",
+      default: false,
+      description: "Whether due date should be sourced from due_date_fixed or from milestones (in GitLab 11.3 and later)",
+      optional: true,
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:257]: Same comment here. If it is `optional` the `default` shouldn't be set

File: components/gitlab/gitlab.app.mjs [hunk 3]
```diff
@@ -168,6 +217,45 @@ export default {
       optional: true,
       default: 100,
     },
+    color: {
+      type: "string",
+      label: "Color",
+      description: "The color of the epic. Introduced in GitLab 14.8, behind a feature flag named epic_highlight_color (disabled by default)",
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:223]:
```suggestion
      description: "The color of the epic. Introduced in GitLab 14.8, behind a feature flag named `epic_highlight_color` (disabled by default)",
```

File: components/gitlab/gitlab.app.mjs [hunk 4]
```diff
@@ -168,6 +217,45 @@ export default {
...
+      type: "boolean",
+      label: "Confidential",
+      description: "Whether the epic should be confidential",
+      default: true,
+      optional: true,
+    },
+    created_at: {
+      type: "string",
+      label: "Created at",
+      description: "When the epic was created. Date time string, ISO 8601 formatted, for example 2016-03-11T03:45:40Z . Requires administrator or project/group owner privileges (available in GitLab 13.5 and later)",
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:236]:
```suggestion
      description: "When the epic was created. Date time string, ISO 8601 formatted, for example `2016-03-11T03:45:40Z`. Requires administrator or project/group owner privileges (available in GitLab 13.5 and later)",
```

File: components/gitlab/gitlab.app.mjs [hunk 5]
```diff
@@ -168,6 +217,45 @@ export default {
...
+      optional: true,
+    },
+    updated_at: {
+      type: "string",
+      label: "Updated at",
+      description: "When the epic was updated. Date time string, ISO 8601 formatted, for example 2016-03-11T03:45:40Z . Requires administrator or project/group owner privileges (available in GitLab 13.5 and later)",
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:242]:
```suggestion
      description: "When the epic was updated. Date time string, ISO 8601 formatted, for example `2016-03-11T03:45:40Z`. Requires administrator or project/group owner privileges (available in GitLab 13.5 and later)",
```

File: components/gitlab/gitlab.app.mjs [hunk 6]
```diff
@@ -168,6 +217,45 @@ export default {
...
+      optional: true,
+    },
+    start_date_is_fixed: {
+      type: "boolean",
+      label: "Start date is fixed",
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:247]:
```suggestion
      label: "Start Date Is Fixed",
```

File: components/gitlab/gitlab.app.mjs [hunk 7]
```diff
@@ -168,6 +217,45 @@ export default {
...
+      default: false,
+      description: "Whether start date should be sourced from start_date_fixed or from milestones (in GitLab 11.3 and later)",
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:249]:
```suggestion
      description: "Whether start date should be sourced from **Start Date Is Fixed** or from milestones (in GitLab 11.3 and later)",
```

File: components/gitlab/gitlab.app.mjs [hunk 8]
```diff
@@ -168,6 +217,45 @@ export default {
...
+      optional: true,
+    },
+    due_date_is_fixed: {
+      type: "boolean",
+      label: "Due date is fixed",
```
Comment (jcortes) [inline components/gitlab/gitlab.app.mjs:254]: Please can  you  capitalize all these label text properties?
```suggestion
      label: "Due Date Is Fixed",
```

**Review by jcortes — DISMISSED** [review]
Hi @luancazarine Just added a few minor request changes and observations. thanks

**Review by jcortes — COMMENTED** [review]
Hi @luancazarine LGTM! Ready for QA!

**Pull request merged**